### PR TITLE
device: Improve dialog text

### PR DESCRIPTION
**Pull request by A6GibKm**
See https://gitlab.gnome.org/GNOME/xdg-desktop-portal-gnome/-/issues/35.

I did my best but I was not able to test this in action.

### CODE REVIEW

**Comment by A6GibKm** [comment]
**cc** @allanday

**Comment by A6GibKm** [comment]
Should this use subtitle rather than body?

File: src/device.c [hunk 1]
```diff
@@ -113,40 +112,37 @@ device_query_permission_sync (const char *app_id,
...
-            subtitle = g_strdup (_("An application wants to use your microphone."));
-          else
-            subtitle = g_strdup_printf (_("%s wants to use your microphone."), g_app_info_get_display_name (info));
+          if (info) {
+            title = g_strdup_printf (_("Allow %s to Use the Microphone?"), g_app_info_get_display_name (info));
+            body = g_strdup_printf (_("%s wants to access recording devices."), g_app_info_get_display_name (info));
+          } else {
+            title = g_strdup (_("Allow app to Use the Microphone?"));
+            body = g_strdup (_("An app wants to access recording devices."));
+          }
```
Comment (GeorgesStavracas) [inline src/device.c:121]: Please follow the code style of the project

File: src/device.c [hunk 2]
```diff
@@ -113,40 +112,37 @@ device_query_permission_sync (const char *app_id,
...
-            subtitle = g_strdup (_("An application wants to play sound."));
-          else
-            subtitle = g_strdup_printf (_("%s wants to play sound."), g_app_info_get_display_name (info));
+          if (info) {
+            title = g_strdup_printf (_("Allow %s to Use the Speakers?"), g_app_info_get_display_name (info));
+            body = g_strdup_printf (_("%s wants to access audio devices."), g_app_info_get_display_name (info));
+          } else {
+            title = g_strdup (_("Allow app to Use the Speakers?"));
+            body = g_strdup (_("An app wants to access audio devices."));
+          }
```
Comment (GeorgesStavracas) [inline src/device.c:133]: Please follow the code style of the project

File: src/device.c [hunk 3]
```diff
@@ -113,40 +112,37 @@ device_query_permission_sync (const char *app_id,
...
-            subtitle = g_strdup (_("An application wants to use your camera."));
-          else
-            subtitle = g_strdup_printf (_("%s wants to use your camera."), g_app_info_get_display_name (info));
+          if (info) {
+            title = g_strdup_printf (_("Allow %s to Use the Camera?"), g_app_info_get_display_name (info));
+            body = g_strdup_printf (_("%s wants to access camera devices."), g_app_info_get_display_name (info));
+          } else {
+            title = g_strdup (_("Allow app to Use the Camera?"));
+            body = g_strdup (_("An app wants to access camera devices."));
+          }
```
Comment (GeorgesStavracas) [inline src/device.c:145]: Please follow the code style of the project

**Review by GeorgesStavracas — CHANGES_REQUESTED** [review]
Can you share a screenshot of how it looks like with your changes? It would ease design review.

**Comment by GeorgesStavracas** [comment]
To test this, you can run `src/xdg-desktop-portal --replace`

This PR closes https://github.com/flatpak/xdg-desktop-portal/issues/1042

**Comment by A6GibKm** [comment]
> To test this, you can run `src/xdg-desktop-portal --replace`
> 
> This PR closes #1042

Unfortunately, that does not work on my system.  For some reason portals are broken after building xdp and running that on my host. I have no idea why

```
$ G_MESSAGES_DEBUG=all _build/src/xdg-desktop-portal --replace
(_build/src/xdg-desktop-portal:96645): xdg-desktop-portal-DEBUG: 21:32:26.666: Looking for portals configuration in '/var/home/deathwish/.config/xdg-desktop-portal/gnome-portals.conf'
(_build/src/xdg-desktop-portal:96645): xdg-desktop-portal-DEBUG: 21:32:26.667: Looking for portals configuration in '/var/home/deathwish/.config/xdg-desktop-portal/portals.conf'
(_build/src/xdg-desktop-portal:96645): xdg-desktop-portal-DEBUG: 21:32:26.667: Looking for portals configuration in '/etc/xdg/xdg-desktop-portal/gnome-portals.conf'
(_build/src/xdg-desktop-portal:96645): xdg-desktop-portal-DEBUG: 21:32:26.667: Looking for portals configuration in '/etc/xdg/xdg-desktop-portal/portals.conf'
(_build/src/xdg-desktop-portal:96645): xdg-desktop-portal-DEBUG: 21:32:26.667: Looking for portals configuration in '/usr/local/etc/xdg-desktop-portal/gnome-portals.conf'
(_build/src/xdg-desktop-portal:96645): xdg-desktop-portal-DEBUG: 21:32:26.667: Looking for portals configuration in '/usr/local/etc/xdg-desktop-portal/portals.conf'
(_build/src/xdg-desktop-portal:96645): xdg-desktop-portal-DEBUG: 21:32:26.667: Looking for portals configuration in '/var/home/deathwish/.local/share/xdg-desktop-portal/gnome-portals.conf'
(_build/src/xdg-desktop-portal:96645): xdg-desktop-portal-DEBUG: 21:32:26.667: Looking for portals configuration in '/var/home/deathwish/.local/share/xdg-desktop-portal/portals.conf'
(_build/src/xdg-desktop-portal:96645): xdg-desktop-portal-DEBUG: 21:32:26.667: Looking for portals configuration in '/var/home/deathwish/.local/share/flatpak/exports/share/xdg-desktop-portal/gnome-portals.conf'
(_build/src/xdg-desktop-portal:96645): xdg-desktop-portal-DEBUG: 21:32:26.667: Looking for portals configuration in '/var/home/deathwish/.local/share/flatpak/exports/share/xdg-desktop-portal/portals.conf'
(_build/src/xdg-desktop-portal:96645): xdg-desktop-portal-DEBUG: 21:32:26.667: Looking for portals configuration in '/var/lib/flatpak/exports/share/xdg-desktop-portal/gnome-portals.conf'
(_build/src/xdg-desktop-portal:96645): xdg-desktop-portal-DEBUG: 21:32:26.667: Looking for portals configuration in '/var/lib/flatpak/exports/share/xdg-desktop-portal/portals.conf'
(_build/src/xdg-desktop-portal:96645): xdg-desktop-portal-DEBUG: 21:32:26.667: Looking for portals configuration in '/usr/local/share/xdg-desktop-portal/gnome-portals.conf'
(_build/src/xdg-desktop-portal:96645): xdg-desktop-portal-DEBUG: 21:32:26.667: Looking for portals configuration in '/usr/local/share/xdg-desktop-portal/portals.conf'
(_build/src/xdg-desktop-portal:96645): xdg-desktop-portal-DEBUG: 21:32:26.667: Looking for portals configuration in '/usr/share/xdg-desktop-portal/gnome-portals.conf'
(_build/src/xdg-desktop-portal:96645): xdg-desktop-portal-DEBUG: 21:32:26.667: load portals from /usr/local/share/xdg-desktop-portal/portals
(_build/src/xdg-desktop-portal:96645): GLib-GIO-DEBUG: 21:32:26.676: Using cross-namespace EXTERNAL authentication (this will deadlock if server is GDBus < 2.73.3)
(_build/src/xdg-desktop-portal:96645): GLib-GIO-DEBUG: 21:32:26.678: _g_io_module_get_default: Found default implementation local (GLocalVfs) for ‘gio-vfs’
(_build/src/xdg-desktop-portal:96645): GLib-GIO-DEBUG: 21:32:26.678: Using cross-namespace EXTERNAL authentication (this will deadlock if server is GDBus < 2.73.3)
(_build/src/xdg-desktop-portal:96645): GLib-GIO-DEBUG: 21:32:26.682: Failed to initialize portal (GMemoryMonitorPortal) for gio-memory-monitor: Not using portals
(_build/src/xdg-desktop-portal:96645): GLib-GIO-DEBUG: 21:32:26.682: _g_io_module_get_default: Found default implementation dbus (GMemoryMonitorDBus) for ‘gio-memory-monitor’
(_build/src/xdg-desktop-portal:96645): xdg-desktop-portal-DEBUG: 21:32:26.682: providing portal org.freedesktop.portal.MemoryMonitor
(_build/src/xdg-desktop-portal:96645): GLib-GIO-DEBUG: 21:32:26.682: Failed to initialize portal (GPowerProfileMonitorPortal) for gio-power-profile-monitor: Not using portals
(_build/src/xdg-desktop-portal:96645): GLib-GIO-DEBUG: 21:32:26.682: Using cross-namespace EXTERNAL authentication (this will deadlock if server is GDBus < 2.73.3)
(_build/src/xdg-desktop-portal:96645): GLib-GIO-DEBUG: 21:32:26.682: _g_io_module_get_default: Found default implementation dbus (GPowerProfileMonitorDBus) for ‘gio-power-profile-monitor’
(_build/src/xdg-desktop-portal:96645): xdg-desktop-portal-DEBUG: 21:32:26.682: providing portal org.freedesktop.portal.PowerProfileMonitor
(_build/src/xdg-desktop-portal:96645): GLib-GIO-DEBUG: 21:32:26.682: Failed to initialize portal (GNetworkMonitorPortal) for gio-network-monitor: Not using portals
(_build/src/xdg-desktop-portal:96645): GLib-GIO-DEBUG: 21:32:26.684: _g_io_module_get_default: Found default implementation networkmanager (GNetworkMonitorNM) for ‘gio-network-monitor’
(_build/src/xdg-desktop-portal:96645): xdg-desktop-portal-DEBUG: 21:32:26.684: providing portal org.freedesktop.portal.NetworkMonitor
(_build/src/xdg-desktop-portal:96645): GLib-GIO-DEBUG: 21:32:26.684: _g_io_module_get_default: Found default implementation dconf (DConfSettingsBackend) for ‘gsettings-backend’
(_build/src/xdg-desktop-portal:96645): dconf-DEBUG: 21:32:26.684: watch_fast: "/system/proxy/" (establishing: 0, active: 0)
(_build/src/xdg-desktop-portal:96645): dconf-DEBUG: 21:32:26.684: watch_fast: "/system/proxy/http/" (establishing: 0, active: 0)
(_build/src/xdg-desktop-portal:96645): dconf-DEBUG: 21:32:26.684: watch_fast: "/system/proxy/https/" (establishing: 0, active: 0)
(_build/src/xdg-desktop-portal:96645): dconf-DEBUG: 21:32:26.684: watch_fast: "/system/proxy/ftp/" (establishing: 0, active: 0)
(_build/src/xdg-desktop-portal:96645): dconf-DEBUG: 21:32:26.684: watch_fast: "/system/proxy/socks/" (establishing: 0, active: 0)
(_build/src/xdg-desktop-portal:96645): GLib-GIO-DEBUG: 21:32:26.685: _g_io_module_get_default: Found default implementation gnome (GProxyResolverGnome) for ‘gio-proxy-resolver’
(_build/src/xdg-desktop-portal:96645): xdg-desktop-portal-DEBUG: 21:32:26.685: providing portal org.freedesktop.portal.ProxyResolver
(_build/src/xdg-desktop-portal:96645): xdg-desktop-portal-DEBUG: 21:32:26.685: providing portal org.freedesktop.portal.Trash
(_build/src/xdg-desktop-portal:96645): dconf-DEBUG: 21:32:26.685: watch_established: "/system/proxy/" (establishing: 1)
(_build/src/xdg-desktop-portal:96645): dconf-DEBUG: 21:32:26.685: watch_established: "/system/proxy/http/" (establishing: 1)
(_build/src/xdg-desktop-portal:96645): dconf-DEBUG: 21:32:26.686: watch_established: "/system/proxy/https/" (establishing: 1)
(_build/src/xdg-desktop-portal:96645): xdg-desktop-portal-DEBUG: 21:32:26.686: providing portal org.freedesktop.portal.GameMode
(_build/src/xdg-desktop-portal:96645): dconf-DEBUG: 21:32:26.686: watch_established: "/system/proxy/ftp/" (establishing: 1)
(_build/src/xdg-desktop-portal:96645): dconf-DEBUG: 21:32:26.686: watch_established: "/system/proxy/socks/" (establishing: 1)
(_build/src/xdg-desktop-portal:96645): xdg-desktop-portal-DEBUG: 21:32:26.688: providing portal org.freedesktop.portal.Realtime

(_build/src/xdg-desktop-portal:96645): xdg-desktop-portal-WARNING **: 21:32:26.688: No skeleton to export
(_build/src/xdg-desktop-portal:96645): xdg-desktop-portal-DEBUG: 21:32:26.688: org.freedesktop.portal.Desktop acquired

```

**Comment by GeorgesStavracas** [comment]
You probably need to restart `xdg-desktop-portal-gnome` after replacing `xdg-desktop-portal`

**Pull request merged**